src/any_llm/providers/bedrock/utils.py

```
@@ -38,24 +46,115 @@
+    if not tools:
+        return
+    for tool in tools:
+        if tool.get("function", {}).get("name") == _STRUCTURED_OUTPUT_TOOL_NAME:
+            msg = (
+                f"Tool name '{_STRUCTURED_OUTPUT_TOOL_NAME}' is reserved by any-llm for "
+                "emulating response_format on bedrock and cannot be used as a tool name."
+            )
+            raise InvalidRequestError(msg, provider_name="bedrock")
+
+
 def _convert_params(params: CompletionParams, kwargs: dict[str, Any]) -> dict[str, Any]:
```

Ruff (0.16.1)

[warning] Too many branches (13 > 12)

(PLR0912)